src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -105,28 +105,11 @@ internal class SourcePropertyAccessorSymbol : SourceMemberMethodSymbol
                 modifiers: new SyntaxTokenList(),
                 methodKind,
                 usesInit,
-                isAutoPropertyAccessor: true,
```

Comment:
I think we could create a new class, `SynthesizedRecordPropertyAccessorSymbol`, which could be an inner private class to `SynthesizedRecordPropertySymbol`, and remove this method from here. @AlekseyTs Does this suggestion look good? Should I go for it in this PR or a separate PR?

Reply:
>I think we could create a new class, ```SynthesizedRecordPropertyAccessorSymbol``, which could be an inner private class to ```SynthesizedRecordPropertySymbol```, and remove this method from here. Does this suggestion look good? Should I go for it in this PR or a separate PR?

What benefits do you see from introducing that new class? Will it have special behavior for some scenarios? Would simply moving the method to ``````SynthesizedRecordPropertySymbol``` work?

Reply:
@AlekseyTs Moving the method will require exposing some constructor.

Reply:
I wouldn't introduce a new class just to move the method.

Reply:
So should I move it and make the constructor `internal` or keep it as it is?

Reply:
> So should I move it and make the constructor ``internal``` or keep it as it is?

Let's keep it as is. There is really not much difference if we move it. We will still need a special entry point here, the exposed constructor, and letting everyone access it doesn't feel like a change to the better.